dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -278,6 +294,49 @@ private TrackerTypeReport handleEvents( Session session, TrackerBundle bundle )
         return typeReport;
     }
 
+    private TrackerTypeReport handleRelationships( Session session, TrackerBundle bundle )
+    {
+        List<Relationship> relationships = bundle.getRelationships();
+        TrackerTypeReport typeReport = new TrackerTypeReport( TrackerType.RELATIONSHIP );
+
+        relationships.forEach( o -> bundleHooks.forEach( hook -> hook.preCreate( Relationship.class, o, bundle ) ) );
+        session.flush();
```

Comment:
Explicit session flush is quite extreme? Can you add a comment why it is required?

Reply:
Removing the flushing but the one that is called when the user select the option to flush every time and object is added.